packages/vitest/src/node/coverage.ts

```diff
@@ -326,7 +340,11 @@ export class BaseCoverageProvider<Options extends ResolvedCoverageOptions<'istan
 
     // Remove empty reports directory, e.g. when only text-reporter is used
     if (readdirSync(this.options.reportsDirectory).length === 0) {
-      await fs.rm(this.options.reportsDirectory, { recursive: true })
+      try {
+        await fs.rm(this.options.reportsDirectory, { recursive: true })
+      }
+      catch {
+      }
```

Comment:
We want to throw the errors from failed `fs.rm` here.

Reply:
Why do you care if you cannot delete an empty directory? Should it fail the whole operation?

Reply:
Users do care: https://github.com/vitest-dev/vitest/issues/5722

We should be able to catch regression if we break this functionality.